lib/logstash/outputs/http.rb

```diff
@@ -89,31 +89,29 @@ def register
     validate_format!
   end # def register
 
-  # TODO: (colin) the request call sequence + async handling will have to be reworked when using
-  # Maticore >= 5.0. I will set a version constrain in the gemspec for this.
-  def receive(event)
+  def multi_receive(events)
+    events.each {|event| receive(event, :parallel)}
+    client.execute!
+  end
+
+  # Once we no longer need to support Logstash < 2.2 (pre-ng-pipeline)
+  # We don't need to handle :background style requests
```

Comment:
I would appreciate a more detailed information on what request_proxy_type background/parallel actually does.
the name `request_proxy_type` makes me think of types of "request proxies"? *feels lost and confused*